### PR TITLE
algorithm: unique paths

### PR DESCRIPTION
**Pull request by hiitesh1127**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

`This Problem is the most common pattern of Dynamic Programming`
### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1211"></a>

### CODE REVIEW

File: Dynamic-Programming/UniquePaths.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @param {number} m
+ * @param {number} n
+ * @return {number}
+ */
+
+const uniquePaths = (m, n) => {
+  // only one way to reach end
+  if (m === 1 || n === 1) return 1
+
+  // build a mxn grid
```
Comment (appgurueu) [inline Dynamic-Programming/UniquePaths.js:26]: Note that you don't even need the full grid; keeping just the last row & the last value suffices.

File: Dynamic-Programming/UniquePaths.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const paths = []
+  for (let i = 0; i < n; i++) paths.push([...new Array(m).fill(1)])
+
+  // base case, position 1x1 has only 1 move
+  paths[1][1] = 1
+
+  for (let i = 1; i < n; i++) {
+    for (let j = 1; j < m; j++) {
+      paths[i][j] = paths[i - 1][j] + paths[i][j - 1]
```
Comment (appgurueu) [inline Dynamic-Programming/UniquePaths.js:35]: Consider adding a comment explaining the rationale behind this (something like "robot may reach (i, j) by going either right or down")

File: Dynamic-Programming/tests/UniquePaths.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { uniquePaths } from '../UniquePaths'
+
+describe('Unique Paths', () => {
+  it('should return 28 when m is 3 and n is 7', () => {
+    const data = uniquePaths(3, 7)
```
Comment (appgurueu) [inline Dynamic-Programming/tests/UniquePaths.test.js:5]: Just inline this, this variable is not needed

File: Dynamic-Programming/tests/UniquePaths.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    expect(data).toBe(28)
+  })
+
+  it('should return 48620 when m is 10 and n is 10', () => {
+    const data = uniquePaths(10, 10)
```
Comment (appgurueu) [inline Dynamic-Programming/tests/UniquePaths.test.js:10]: same here

File: Dynamic-Programming/UniquePaths.js (hunk 1 above)
Comment (hiitesh1127) [reply Dynamic-Programming/UniquePaths.js]: Thanks for this suggestion

**Review by appgurueu — APPROVED** [review]

**Comment by hiitesh1127** [comment]
Please review @raklaptudirm

**Pull request merged**